feat: prepare UserCell to show verification statuses [WPB-765]

caldrian opened the pull request:
<!--do not remove this marker, its needed to replace info when ticket title is updated -->
<!--jira-description-action-hidden-marker-start-->

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-765" title="WPB-765" target="_blank"><img alt="Story" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10815?size=medium" />WPB-765</a>  [iOS] 5.3.16 Indicate verified user
  </td></table>
  <br />
 

<!--jira-description-action-hidden-marker-end-->
<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Adds the possibility to show a second verification shield after user's names as attributed string attachments.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

caldrian committed:
chore: move Avatar into UserStatus

caldrian committed:
chore: remove UIImageView extension

caldrian committed:
Merge remote-tracking branch 'github/develop' into feat/use-userstatus-for-other-users-WPB-765

# Conflicts:
#	wire-ios/Wire-iOS/Sources/UserInterface/Components/Views/UserCell.swift

caldrian committed:
Chore: Try to add preview for UserCell

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into feat/use-userstatus-for-other-users-WPB-765

caldrian committed:
chore: display generic cell

caldrian committed:
chore: try to not inject users into views

caldrian committed:
chore: move Avatar to UserStatus

caldrian committed:
fix: use gesture recognizer instead of inheritance from UIControl

caldrian committed:
chore: prepare for making views passive

caldrian committed:
refactor: minor changes

caldrian committed:
fix: split files

caldrian committed:
chore: split more files

caldrian committed:
Merge branch 'develop' into feat/passive-avatarimageview-WPB-765

caldrian committed:
refactor: minor changes

caldrian committed:
Merge branch 'feat/passive-avatarimageview-WPB-765' into feat/use-userstatus-for-other-users-WPB-765

# Conflicts:
#	wire-ios/Wire-iOS/Sources/UserInterface/Availability/UserStatus.swift
#	wire-ios/Wire-iOS/Sources/UserInterface/Components/AvatarImageView.swift
#	wire-ios/Wire-iOS/Sources/UserInterface/Components/UserImageView.swift
#	wire-ios/Wire-iOS/Sources/UserInterface/SelfProfile/AccountViews/BaseAccountView.swift

caldrian committed:
chore: rename method

caldrian committed:
chore: allow updating UserCell without selfUser instance

caldrian committed:
fix: try to make preview for collection view cells

caldrian committed:
chore: remove preview of UserCell

caldrian committed:
chore: change argument type of configure method

caldrian committed:
chore: create preview for AvailabilityStringBuilder

caldrian committed:
fix: updating title

caldrian committed:
fix: user status not being visible

caldrian committed:
chore: pass userStatus value to UserCell.configure

caldrian committed:
chore: pass verification status via userStatus to UserCell

caldrian committed:
chore: create UserStatus from UserType

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into feat/use-userstatus-for-other-users-WPB-765

caldrian committed:
fix: undo some changes

caldrian committed:
fix: move Avatar out of UserStatus

caldrian committed:
chore: create UserStatusViewControllerDelegate

caldrian committed:
chore: don't inject user into UserStatusViewController

caldrian committed:
chore: support trailing attachments in IconStringBuilder

caldrian committed:
chore: show shields as trailing icons

caldrian commented on wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/ConversationListTopBarView.swift:
```diff
@@ -0,0 +1,155 @@
+
+import UIKit
+
+final class TopBar: UIView {
```
I extracted this class from [ConversationListTopBarView.swift](https://github.com/wireapp/wire-ios/pull/979/files#diff-85758b93c193f5ea5105f37e67b9f357f2671567e71efce25b0fe1666cb745b3).

caldrian commented on wire-ios/Wire-iOS/Sources/UserInterface/ConversationList/ConversationListTopBarViewController.swift:
```diff
@@ -288,135 +315,3 @@ extension UIView {
     }
 }
-
-final class TopBar: UIView {
```
I moved this class into a separate file.

caldrian committed:
fix: cleanup and minor corrections

caldrian committed:
chore: add todo

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into feat/use-userstatus-for-other-users-WPB-765

caldrian committed:
fix: add vertical offset for verification images

caldrian committed:
Merge branch 'develop' into feat/use-userstatus-for-other-users-WPB-765

caldrian committed:
fix: remove some ToDos

caldrian committed:
fix: line break mode

caldrian committed:
fix: pass all arguments to UserStatus

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into feat/use-userstatus-for-other-users-WPB-765

caldrian committed:
chore: add convenience initializer for UserStatus

caldrian committed:
fix: add a doc string

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into feat/use-userstatus-for-other-users-WPB-765

caldrian committed:
fix: doc string

caldrian committed:
fix: accessibility label for user status view

caldrian committed:
refactor: code for generating text attachments in UserStatusView

caldrian committed:
fix: showing duplicate verification shields

caldrian committed:
chore: update snapshot reference images

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into feat/use-userstatus-for-other-users-WPB-765

markusfassbender commented on wire-ios-transport/Source/TransportEncoding/Date+TransportCoding.swift:
```diff
@@ -26,6 +26,9 @@ private let iso8601DateWithFractionalSecondsFormatter = {
 }()
 
 /// Covers the cases where no fractional seconds are provided.
+/// Due to a bug in the backend timestamps without fractional seconds are received.
+/// [WPB-6529](https://wearezeta.atlassian.net/browse/WPB-6529)
+/// After the bug is fixed, this date formatter can be removed.
 private let iso8601DateFormatter = {
```
👍

markusfassbender commented on wire-ios/Wire-iOS/Sources/UserInterface/Availability/AvailabilityStringBuilder.swift:
```diff
@@ -16,24 +16,31 @@
 import WireDataModel
 import WireCommonComponents
 
-final class AvailabilityStringBuilder: NSObject {
+enum AvailabilityStringBuilder {
```
😍

markusfassbender commented on wire-ios/Wire-iOS/Sources/UserInterface/Availability/TitleView.swift:
```diff
@@ -24,7 +24,7 @@ class TitleView: UIView, DynamicTypeCapable {
     var titleColor: UIColor?
     var titleFont: FontSpec?
-    var tapHandler: ((UIButton) -> Void)?
+    var tapHandler: (UIButton) -> Void = { _ in }
```
What is the advantage of having this filled with empty implementation? For me it's usually clearer to see `tapHandler = nil` when debugging for example.

caldrian replied:
I guess there's manly the disadvantage you mentioned. 
I just prefer to use a default value instead of a redundant optional. (e.g. empty string instead of `nil`, no-op closure instead of `nil` etc.)

markusfassbender commented on wire-ios/WireCommonComponents/UIColor+AccentColor.swift:
```diff
@@ -34,12 +34,16 @@ public enum AccentColor: Int16, CaseIterable {
         return AccentColor.allSelectable().randomElement()!
     }
 
+    public init() {
+        self = .allCases[0]
+    }
```
To read this from some place, I find this more implicit and one needs to look up what `= AccentColor()` case is chosen compared to `= .blue`.

caldrian replied:
Also here I wanted to offer a default initialiser. I'll replace it by a static `default` property.

markusfassbender commented on wire-ios-data-model/Source/Model/User/Availability/Availability.swift:
```diff
@@ -19,4 +19,8 @@
 @objc(ZMAvailability)
 public enum Availability: Int, CaseIterable {
     case none, available, busy, away
+
+    public init() {
+        self = .none
+    }
```
To read this from some place, I find this more implicit and one needs to look up what `= AccentColor()` case is chosen compared to `= .none`.

caldrian replied:
I wanted to offer an initialiser like any other primitive has, e.g. `Bool()`, `String()`, `Int()` etc.
But I'll revert it.

markusfassbender reviewed (approved):
Great job, I've commented some nitpicks.

caldrian committed:
fix: revert unclear changes

caldrian committed:
chore: add comment

caldrian committed:
chore: add section names

# Conflicts:
#	wire-ios-data-model/Source/Model/Conversation/ConversationVerificationStatus.swift
#	wire-ios/Wire-iOS/Sources/UserInterface/GroupDetails/GroupConversationVerificationStatusView.swift

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into feat/use-userstatus-for-other-users-WPB-765

caldrian commented on wire-ios/Wire-iOS/Sources/UserInterface/SelfProfile/AccountViews/BaseAccountView.swift:
```diff
@@ -237,12 +159,32 @@ class BaseAccountView: UIView {
+    case accountSelector
+}
+
+// MARK: -
```
Sometimes I prefer to have just a line without a title, but I added text now.
